### DISCUSSION TITLE
Supported database system

### DISCUSSION THREAD

**Discussion by DorianCoding** [stack-auth/stack-auth#355] [General]
Hello,

On the docs it seems only postgres database are supported on hosted version. But Prisma supports a lot of databases system. Can we use all of them and if not, which one are available? And is there any plans to add others?

Thanks.

**Comment by fomalhautb** [comment]
We currently don't use any Postgres-specific features so it probably works with other DBs, we have never tested them though. We currently don't have plans to add other DBs as the users can use a different DB for their project, it is not limited by the DB choice of Stack Auth

**Comment by DorianCoding** [comment]
Thanks for your answer.

However on [schema.prisma](https://github.com/stack-auth/stack/blob/dev/apps/backend/prisma/schema.prisma) is:
```
datasource db {
  provider  = "postgresql"
  url       = env("STACK_DATABASE_CONNECTION_STRING")
  directUrl = env("STACK_DIRECT_DATABASE_CONNECTION_STRING")
}
```

And [Prism stated](https://www.prisma.io/docs/orm/reference/prisma-schema-reference#datasource):

```
The following providers are available:

    [sqlite](https://www.prisma.io/docs/orm/overview/databases/sqlite)
    [postgresql](https://www.prisma.io/docs/orm/overview/databases/postgresql)
    [mysql](https://www.prisma.io/docs/orm/overview/databases/mysql)
    [sqlserver](https://www.prisma.io/docs/orm/overview/databases/sql-server)
    [mongodb](https://www.prisma.io/docs/orm/overview/databases/mongodb)
    [cockroachdb](https://www.prisma.io/docs/orm/overview/databases/cockroachdb)
```

So if any DB presented can be used, would it be possible to overwrite the datasource in the env file with a `STACK_DATABASE_PROVIDER`? Because I don't achieve to access my mysql database just by changing the URL using the docker component.

Thanks.

**Reply by madster456** [reply]
> We currently don't use any Postgres-specific features so it probably works with other DBs, we have never tested them though. We currently don't have plans to add other DBs as the users can use a different DB for their project, it is not limited by the DB choice of Stack Auth

Reiterating here. If you would like to open a feature request, take a look at our [Feedback Board](https://feedback.stack-auth.com/).

Let me know if you have any more questions!

---

<img src="https://stack-auth.com/_next/static/media/stack-auth-logo.84c18f5e.svg" width="200" />
